src/tools/gophish_export.py

```diff
@@ -253,6 +253,44 @@ def get_application(rawEvent):
     return application
 
 
+def export_user_reports(api, assessment_id):
+    """Build and export a user_report JSON object for each campaign in an assessment."""
+    campaign_ids = get_campaign_ids(api, assessment_id)
+
+    for campaign_id in campaign_ids:
+        first_report = None
+        user_report_doc = dict()
+        campaign = get_campaign_data(api, campaign_id)
+
+        # iterate over clicks and find the earliest click
+        for click in campaign["clicks"]:
+            if (
+                first_report is None
+                or datetime.strptime(click["time"].split(".")[0], "%Y-%m-%dT%H:%M:%S")
+                < first_report
+            ):
+                first_report = datetime.strptime(
+                    click["time"].split(".")[0], "%Y-%m-%dT%H:%M:%S"
+                )
```

Comment:
Here's a slightly cleaner way to write this:
```suggestion
            click_time = datetime.strptime(
                click["time"].split(".")[0], "%Y-%m-%dT%H:%M:%S"
            )
            if first_report is None or click_time < first_report:
                first_report = click_time
```

Reply:
See 724b35ac39da24c61713cc0dd0addb7a5f3befe5